src/v/security/gssapi_authenticator.cc

```diff
@@ -136,6 +136,11 @@ class gssapi_authenticator::impl {
         return _rp_user_principal;
     }
 
+    void reset() {
+        _context.reset();
+        _server_creds.reset();
+    }
```

Comment:
Should there be a defined dtor that calls this as well?

Reply:
The destructors of the objects call reset() on themselves.